queries/c_sharp/indents.scm

```diff
@@ -0,0 +1,117 @@
+[
+ (property_declaration)
+ ; (field_declaration)
+ (variable_declaration)
+ (accessor_declaration)
+
+ ; (declaration_list)
+
+ (parameter_list)
+ (type_parameter_list)
+ (accessor_list)
+ (attribute_argument_list)
+ (base_list)
+ (argument_list)
+
+ (binary_expression)
+ (expression_statement)
+ (return_statement)
+ (local_declaration_statement)
+ (local_function_statement)
+ (yield_statement)
+ (fixed_statement)
+ (unsafe_statement)
+ (labeled_statement)
+ (for_each_statement)
+ (for_statement)
+ (if_statement)
+ (switch_expression)
+ (switch_section)
+ (switch_statement)
+ (while_statement)
+ (lambda_expression)
+ (do_statement)
+ (checked_statement)
+ (try_statement)
+ (using_statement)
+ (local_function_statement)
+ (object_creation_expression)
+ (array_creation_expression)
+ (implicit_object_creation_expression)
+
+ ] @indent.begin
+
+(method_declaration body: _ @indent.begin)
+(namespace_declaration body: _ @indent.begin)
+(record_declaration body: _ @indent.begin)
+(class_declaration body: _ @indent.begin)
+(struct_declaration body: _ @indent.begin)
+(constructor_declaration body: _ @indent.begin)
+(interface_declaration body: _ @indent.begin)
+(enum_declaration body: _ @indent.begin)
+
+(property_declaration (accessor_list "{" @indent.branch) @indent.dedent)
+
+(switch_section
+  (case_switch_label) (case_switch_label) @indent.branch)
+
+
+(if_statement (block "{" @indent.branch))
+(if_statement alternative: (if_statement) @indent.dedent)
+
+(while_statement
+  (block "{" @indent.branch))
+
+(array_creation_expression
+  (initializer_expression "{" @indent.branch))
+(object_creation_expression
+  (initializer_expression "{" @indent.branch))
+(implicit_object_creation_expression
+  (initializer_expression "{" @indent.branch))
+
+(for_each_statement 
+  body: (block "{" @indent.branch))
+
+(lambda_expression 
+  body: (block "{" @indent.branch))
+
+(for_statement 
+  body: (block "{" @indent.branch))
+
+(switch_expression "{" @indent.branch)
+
+(switch_statement
+  (switch_body "{" @indent.branch))
+
+(try_statement
+  body: (block "{" @indent.branch))
+(catch_clause
+  body: (block "{" @indent.branch))
+(finally_clause
+  (block "{" @indent.branch))
+
+(fixed_statement 
+  (block "{" @indent.branch))
+
+(using_statement 
+  body: (block "{" @indent.branch))
```

Comment:
I can see a lot of `(parent body: (block ... @capture))` here. Can it be replaced with only `(block ...)`

Reply:
Personal guess: You need this because you capture the parent node with `@indent.begin`, which is probably not needed.

Reply:
I wish I remember the reason. But I clearly do remember that I couldn't make it work properly via `(block) @indent.begin`.

Reply:
At its current state, I fear this is too messy for me, you, future you and others to maintain this.

If you think it's fine for you to dedicate some hours if issues pop up, and that the queries doesn't make editing C# feel sluggish, then I'm okay with it

Reply:
hm, I will try what happens

Reply:
I see why I might do it this way. You may have most of these conditionals and loops not being followed by block, but by plain statement (or expression).

I had the idea to make it something like `(_ body: (block "{" @indent.branch))` to avoid repeating everything with this schema, but I couldn't figure out proper syntax or make it work.